kong/db/schema/typedefs.lua

```diff
@@ -507,18 +522,16 @@ local function validate_path_with_regexes(path)
     return true
   end
 
-  path = path:sub(2)
+  return is_valid_regex(path:sub(2))
+end
 
-  -- the value will be interpreted as a regex by the router; but is it a
-  -- valid one? Let's dry-run it with the same options as our router.
-  local _, _, err = ngx.re.find("", path, "aj")
-  if err then
-    return nil,
-           string.format("invalid regex: '%s' (PCRE returned: %s)",
-                         path, err)
+
+local function validate_regex_or_plain_pattern(pattern)
+  if pattern:sub(1, 1) ~= "~" then
```

Comment:
I would personally extract this to yet another function and we could've ended up with code like this:

```lua
local function is_plain_patter(pattern)
  return pattern:sub(1, 1) ~= "~"
end

local function validate_regex_or_plain_pattern(pattern)
  return is_plain_patter(pattern) or is_valid_regex(pattern:sub(2))
end
```

We could also reuse then `is_plain_patter` in the function above so that there'll one less check against `"~"` character which imho is a bit _magic_stringy_

... but that's probably just nitpicking - I'm happy with the code we have right now 😃

Reply:
+1